components/scream/src/physics/shoc/shoc_energy_fixer_disp.cpp

```diff
@@ -0,0 +1,81 @@
+#include "shoc_functions.hpp"
+
+#include "ekat/kokkos/ekat_subview_utils.hpp"
+
+namespace scream {
+namespace shoc {
+
+template<>
+void Functions<Real,DefaultDevice>
+::shoc_energy_fixer_disp(
+  const Int&                   shcol,
+  const Int&                   nlev,
+  const Int&                   nlevi,
+  const Scalar&                dtime,
+  const Int&                   nadv,
+  const view_2d<const Spack>&  zt_grid,
+  const view_2d<const Spack>&  zi_grid,
+  const Int&                   se_b_slot,
```

Comment:
Maybe we could group these slot indices in a verbose struct, like `shoc::SmallKernelsSlotsIds` or something.